metadata-ingestion/src/datahub/ingestion/source/datahub/datahub_source.py

```diff
@@ -46,43 +50,69 @@ def get_report(self) -> SourceReport:
         return self.report
 
     def get_workunit_processors(self) -> List[Optional[MetadataWorkUnitProcessor]]:
-        return []  # Exactly replicate data from DataHub source
+        # Exactly replicate data from DataHub source
+        return [partial(auto_workunit_reporter, self.get_report())]
 
     def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
-        stop_time = datetime.now(tz=timezone.utc)
-        logger.info(f"Ingesting DataHub metadata up until roughly {stop_time}")
+        self.report.stop_time = datetime.now(tz=timezone.utc)
+        logger.info(f"Ingesting DataHub metadata up until {self.report.stop_time}")
         state = self.stateful_ingestion_handler.get_last_run_state()
-        yield from self._get_mysql_workunits(state.mysql_createdon_datetime, stop_time)
-        self._commit_progress()
-        yield from self._get_kafka_workunits(state.kafka_offsets, stop_time)
-        self._commit_progress()
 
-    def _get_mysql_workunits(
-        self, from_createdon: datetime, stop_time: datetime
+        if self.config.database_connection is not None:
+            yield from self._get_database_workunits(
+                from_createdon=state.database_createdon_datetime
+            )
+            self._commit_progress()
+        else:
+            logger.info(
+                "Skipping ingestion of versioned aspects as no database_connection provided"
+            )
+
+        if self.config.kafka_connection is not None:
+            yield from self._get_kafka_workunits(from_offsets=state.kafka_offsets)
+            self._commit_progress()
+        else:
+            logger.info(
+                "Skipping ingestion of timeseries aspects as no kafka_connection provided"
+            )
+
+    def _get_database_workunits(
+        self, from_createdon: datetime
     ) -> Iterable[MetadataWorkUnit]:
-        logger.info(f"Fetching MySQL aspects from {from_createdon}")
-        reader = DataHubMySQLReader(self.config, self.report)
-        mcps = reader.get_aspects(from_createdon, stop_time)
+        if self.config.database_connection is None:
+            return
```

Comment:
This is repetitive but I prefer to (i) an assertion or (ii) passing `database_connection` as an argument

Reply:
I'm actually fine with an assertion here

Reply:
I don't really like assertions because if we ever make code changes where the assertion is no longer valid, I think it's a really bad error to display to the user. + it bypasses the type system